tests/conftest.py

```diff
@@ -337,6 +337,7 @@ def microvm_factory(request, record_property, results_dir, netns_factory):
     report = request.node.stash[PHASE_REPORT_KEY]
     if "call" in report and report["call"].failed:
         for uvm in uvm_factory.vms:
+            uvm.flush_metrics()
```

Comment:
I don't think this will necessarily work, because dead vms arent removed from `uvm_factors.vms`. We'll probably need to wrap this in a try/except?